server/index.ts

```diff
@@ -55,22 +55,61 @@ import {
 } from "../build/blog.js";
 import { findCurriculumPageBySlug } from "../build/curriculum.js";
 
-async function buildDocumentFromURL(url: string) {
-  const document = Document.findByURL(url);
-  if (!document) {
-    return null;
+async function buildDocumentFromURL(res, url: string) {
```

Comment:
Properly typing the `res` argument here (and everywhere else) exposes some of the return type inconsistency problems:

```suggestion
async function buildDocumentFromURL(res: express.Response, url: string) {
```